draft-handrews-jri.md

```diff
@@ -0,0 +1,541 @@
+
+Depending on the exact functionality offered, the "cache" could be as simple as a look-up table.  It could also perform actual caching tasks such as managing a working set in limited memory, or integrating with a caching system such as one offered by HTTP clients in accordance with {{?RFC9111}}.  JRI does not impose any requirements in this regard.
+
+A JRI cache implementation MUST:
+
+* Be able to associate a document, including non-JRI documents that could be JRI reference targets, with an IRI
+* Locate all JRI identifiers (primary and secondary resources) within any document and automatically include them in the cache
+* Locate all JRI references and resolve any relative IRI-references to full IRIs in the cached representation
+* In its default configuration, treat cache misses as errors, since JRI reference IRIs are assumed to be locators
+* Resolve IRIs that use JSON Pointer fragments as long as the primary resource is in the cache (when such fragments are appropriate for the resource.
+
+JSON Pointer fragment resolution can either be done by indexing valid JSON Pointers, which typically involves understanding JRI location keywords and any similar context-specific locations, or by matching the primary resource and evaluating the JSON Pointer on that resource on demand.  However, discovering JRI identifiers generally needs to be done in advance, as compound documents mean that the location of a primary resource might only be detectable by scanning a document that appears to be a different resource.
+
+A JRI cache implementation SHOULD:
+
+* Keep track of each resource's media type and handle fragments accordingly, unless the cache is implementing a context specification that guarantees consistent media types
+* Respect any cache controls on retrieved resources to avoid burdening servers with superfluous requests; this can often be deferred to a caching network protocol client
+
+A JRI cache implementation MAY:
+
+* Offer a plugin interface for resolving cache misses, keyed off of the IRI scheme or other IRI components
+    * Such a system SHOULD allow plugins that locate resources other than by retrieving the IRI; for example in a development environment it might be preferable to resolve `https:` resources from the local file system
+    * Registering a plugin is sufficient to consider the JRI cache implementation "non-default" with respect to the requirement to treat IRIs as non-locators by default
+
+### Document set transformations
+
+A common family of use cases involves reading one or more JRI-using documents and transforming them in a way that makes them usable by a wider range of tooling.  Note that circular references can cause problems for these use cases, as discussed in {{infinite}}.
```

Comment:
@DavidBiesack I think you're looking at an outdated diff.  Here is the xref that you requested.